### PR TITLE
Add zone resource support

### LINKED ISSUE #22: Add DNS zone resource support
### Methods
- [x] Create a DNS zone in a enf domain `POST /xdns/v1/zones`
     - Request body: 
              ```{"zone_domain_name":"abc.def.xyz", "enf_domain": "2607:6560:23f1::/48", "description":"A DNS domain"}```
- [x] List DNS zones in a enf domain `GET /xdns/v1/zones?enf_domain=<::/48>`
- [x] List a DNS zone in a enf domain (Get a DNS zone) `GET /xdns/v1/zones/{zone_uuid}`
    - Response body:
            ```{
      "id": "b1fc5b2b-05f8-443c-8eb5-ffc4cdb70303",
      "zone_domain_name": "abc.def.xyz",
      "description": "A DNS domain",
      "enf_domain": "2607:6560:23f1::/48",
      "created": "2019-04-26T12:58:59.381Z",
      "modified": null
    }```

- [x] Update a DNS zone in a enf-domain `PUT /xdns/v1/zones/{zone_uuid}`
    - Request body:
          ```{"description":"Updated DNS domain"}```
- [x] Delete a DNS zone in a enf-domain `DELETE /xdns/v1/zones/{zone_uuid}`

### PR DESCRIPTION
**Pull request by dylanirlbeck**
Fixes #22

### CODE REVIEW

File: enf/dns_zone.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+	Description *string `json:"description"`
+}
+
+type zoneResponse struct {
+	Data []*Zone                `json:"data"`
+	Page map[string]interface{} `json:"page"`
+}
+
+// ListDNSZones lists all the DNS zones for a given ENF domain (::/48 address).
+func (s *DNSService) ListDNSZones(ctx context.Context) ([]*Zone, *http.Response, error) {
```
Comment (drbild) [inline enf/dns_zone.go]: This is already on the DNS service, so would just `ListZones` be acceptable?  Ditto for all the other methods.

File: enf/dns_zone.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+	body, resp, err := s.client.put(ctx, path, new(zoneResponse), req)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return body.(*zoneResponse).Data[0], resp, nil
+}
+
+// DeleteDNSZone deletes a zone given its UUID.
+func (s *DNSService) DeleteDNSZone(ctx context.Context, zoneUUID string) (*http.Response, error) {
```
Comment (drbild) [inline enf/dns_zone.go]: `DeleteDNSZone` -> `DeleteZone`.

File: enf/dns_zone_test.go [hunk 3]
```diff
@@ -0,0 +1,205 @@
+package enf
+
+import (
+	"context"
+	"net/http"
+	"testing"
+	"time"
+)
+
+func TestDNSService_ListDNSZones(t *testing.T) {
```
Comment (drbild) [inline enf/dns_zone_test.go]: `TestDNSService_ListDNSZones` -> `TestDNSService_ListZones`.  Ditto for the other test methods.

Comment (dylanirlbeck) [reply enf/dns_zone_test.go]: My bad for missing these. They are fixed in the latest commit.

File: enf/dns_zone_test.go [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+	}
+
+	putTest(testParams)
+}
+
+func TestDNSService_DeleteZone(t *testing.T) {
+	path := "/api/xdns/2019-05-27/zones/1234"
+
+	method := func(client *Client) (interface{}, *http.Response, error) {
+		resp, err := client.DNS.DeleteDNSZone(context.Background(), "1234")
```
Comment (drbild) [inline enf/dns_zone_test.go]: One last one!! :grin:

Comment (dylanirlbeck) [reply enf/dns_zone_test.go]: Oh my haha. My eyes are playing tricks (and I'm rushing a bit). Thanks for the catch!

**Pull request merged**